src/stores/advertises.js

```diff
@@ -71,26 +71,25 @@ export const useAdvertiseStore = defineStore('advertises', {
       if (type) {
         try {
           const q = query(collection(db, 'advertises'), type !== 'All' ? where('status', '==', type) : '', orderBy('created', 'desc'))
-          this._unSubscribe = onSnapshot(q, async (querySnapshot) => {
-            this.setLoaderTrue()
-            let advertises = querySnapshot.docs.map((doc) => {
-              return { id: doc.id, ...doc.data() }
-            })
+          this.setLoaderTrue()
+          const querySnapshot = await getDocs(q)
+          let advertises = querySnapshot.docs.map((doc) => {
+            return { id: doc.id, ...doc.data() }
+          })
 
-            if (!userStore.isAdmin) {
-              advertises = advertises.filter((advertise) => {
-                return advertise.author.id === userStore.getUserId
-              })
-            }
+          if (!userStore.isAdmin) {
+            advertises = advertises.filter((advertise) => {
+              return advertise.author.id === userStore.getUserId
+            })
+          }
 
-            for (const advertise of advertises) {
-              advertise.author = userStore.getUserById(advertise.author.id) || (await userStore.fetchUser(advertise.author.id))
-            }
-            this._advertises = []
-            this.$patch({ _advertises: advertises })
-            await this.computeValues()
-            this.setLoaderFalse()
-          })
+          for (const advertise of advertises) {
+            advertise.author = userStore.getUserById(advertise.author.id) || (await userStore.fetchUser(advertise.author.id))
```

Comment:
And I think for the advertisement, fetching all users for each advertisement is unecessery. 
(To test, remove this for loop and see that nothing changes in the advertisement table, but there are fewer requests)

If we need the user id for ad for editing we could fetch in separately by ad id.